File: code-chef-query.py
from time import time
test_case=int(input())
for test in range(test_case):
    line = input()
    line=line.strip().split(" ")
    k=int(line[1])
    array=list(map(int,input().split()))
    carry=0
    cur=time()
    for index in range(len(array)):
        array[index]+=carry
        if array[index]<k:
            print(index+1)
            break
        else:
            carry=array[index]-k
    if carry>k:
        carry=carry//k
        print(carry+1+len(array))
    print("%.8f" % float(time()-cur))#to check the time taken by the code to execute

File: code_chef_query.c
#include<stdio.h>
int main()
{
    int testcase,size,maxquery,index,carry,days,over;
    scanf("%d",&testcase);
    for(;testcase>0;testcase--)
    {
        scanf("%d %d",&size,&maxquery);
        int query[size];
        for(index=0;index<size;index++)
            scanf("%d",&query[index]);
        carry=0;
        for(index=0;index<size;index++)
        {
            query[index]+=carry;
            if(query[index]<maxquery)
            {
                printf("%d\n",index+1);
                break;            
            }
            else
                carry=query[index]-maxquery;
        }
        if(carry>maxquery)
        {
            carry=carry/maxquery;
            printf("%d\n",carry+size+1);
        }
    }
    return 0;
}
